File: backend-dotnet/src/AlturaNova.Domain/Entities/Product.cs
namespace AlturaNova.Domain.Entities;

/// <summary>A sellable product owned by a vendor. Has one or more variants.</summary>
public class Product
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public Guid VendorId { get; set; }
    public Vendor Vendor { get; set; } = null!;

    public Guid? CategoryId { get; set; }
    public Category? Category { get; set; }

    public string Name { get; set; } = string.Empty;
    public string Handle { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string? ThumbnailUrl { get; set; }
    public string Currency { get; set; } = "EUR";
    public bool IsPublished { get; set; } = true;

    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset UpdatedAt { get; set; } = DateTimeOffset.UtcNow;

    public ICollection<ProductVariant> Variants { get; set; } = new List<ProductVariant>();
    public ICollection<ProductImage> Images { get; set; } = new List<ProductImage>();
}

File: mobile/shared/src/commonMain/kotlin/com/example/ultra/core/domain/model/Product.kt
package com.example.ultra.core.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class Product(
    val id: String,
    val vendorId: String,
    val name: String,
    val description: String,
    val price: Double,
    val oldPrice: Double? = null,
    val currency: String = "NGN",
    val imageUrl: String? = null,
    val thumbnailUrl: String? = null,
    val category: String? = null,
    val inStock: Boolean = true,
    val variants: List<ProductVariant> = emptyList(),
    val handle: String = ""
)
@Serializable
data class ProductVariant(
    val id: String,
    val title: String,
    val sku: String? = null,
    val price: Double = 0.0,
    val currency: String = "EUR",
    val inventoryQuantity: Int = 0
) {
    val isInStock: Boolean get() = inventoryQuantity > 0
}